### PR TITLE
Supply drop event

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Adds a new event for HvH gamemodes, although it fully works in HvX as well.

One faction in the game mode is chosen as the 'sending' faction, and they receive a notification telling them where the supply drop will be landing and when.
image

After a short delay, any other factions are notified of the hostile supply drop as well, giving the friendly faction a bit of a head start. Yes there is a typo there, it's fixed but I'm too lazy to take a new screenshot.
image

When the supply drop is generated, it lands in a random open turf in the target area, and everyone is notified.
image

The supply drop for TGMC and SOM are a randomised supply crate containing medical supplies, or weapons and equipment of various rarities.
image
image

Or if its a funni alien supply drop, it's a larb.
image

As mentioned, the event supports any faction, or even multiple factions, so it's future proofed for gamemodes with new or more factions, and badmins can trigger it safely in HvX modes. Any faction without a specific supply drop will default to the marine one.

changed loot box code a little, they now support multiple rolls on the table if specified.

Oh also added a burstfire SOM shotgun for supply drops only. Because it was satisfying as hell to use when Tye made one, but probs not balanced for general use.
## Why It's Good For The Game
More events are kool
New objectives to fight over in HvH is good
## Changelog
:cl:
add: Adds supply drop event to HvH
add: Added a supply drop specific SOM burst shotgun
/:cl:

### CODE REVIEW

**Comment by FosterGrime-Heymynameslouie** [comment]
holy shit

noodle in the box

a boxed noodle

**Comment by zzurge** [comment]
woah based

File: code/game/objects/items/loot_box.dm [hunk 1]
```diff
@@ -72,8 +77,311 @@
 		/obj/item/weapon/banhammer,
 	)
 
-// 150 to 200 points of value packs, spend 100 points get 150 to 200 in value, basically. Ideally, commons are variety packs, uncommons maybe shake up the round a bit, rares a bit more. Legendaries make the round go wacko. You get a crate of stuff dropped on spawn.
+//Supply drop boxes
+/obj/item/loot_box/supply_drop
+	name = "Supply drop"
```
Comment (XSlayer300) [inline code/game/objects/items/loot_box.dm:82]:
```suggestion
	name = "supply drop"
```

File: code/game/objects/items/loot_box.dm [hunk 2]
```diff
@@ -72,8 +77,311 @@
...
+	desc = "A box of valuable military equipment"
```
Comment (XSlayer300) [inline code/game/objects/items/loot_box.dm:83]:
```suggestion
	desc = "A TGMC-marked box full of valuable military tactical equipment."
```

File: code/game/objects/items/loot_box.dm [hunk 3]
```diff
@@ -72,8 +77,311 @@
...
+		/obj/item/storage/pouch/medical_injectors/medic,
+		/obj/effect/supply_drop/standard_carbine,
+		/obj/effect/supply_drop/standard_rifle,
+		/obj/effect/supply_drop/combat_rifle,
+		/obj/effect/supply_drop/laser_rifle,
+		/obj/effect/supply_drop/standard_shotgun,
+	)
+
+/obj/item/loot_box/supply_drop/som
+	name = "Supply drop"
```
Comment (XSlayer300) [inline code/game/objects/items/loot_box.dm:119]:
```suggestion
	name = "supply drop"
```

File: code/game/objects/items/loot_box.dm [hunk 4]
```diff
@@ -72,8 +77,311 @@
...
+	desc = "A box of valuable SOM military equipment"
```
Comment (XSlayer300) [inline code/game/objects/items/loot_box.dm:120]:
```suggestion
	desc = "A rugged box composed of valuable SOM military materiel."
```

**Comment by Oscari-pog** [comment]
Finally, ayylmaos in HvH

**Review by TiviPlus — CHANGES_REQUESTED** [review]
didnt read through fully but use the qdel intitialize hint instead of qdel

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**